File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = list(s) # 문자열 변경을 위함
    isFirst = True
    for i in range(len(answer)):
        if answer[i].isalpha():
            if isFirst:
                answer[i] = answer[i].upper()
                isFirst = False
            else:
                answer[i] = answer[i].lower()
        elif answer[i] == ' ':
            isFirst = True
        else:
            isFirst = False
    return ''.join(answer)

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.js
function solution(s) {
    const arr = Array.from(s);
    let answer = '';
    let isFirst = true;
    
    arr.forEach((value) => {
        if (value === ' ') { // 공백이면
            answer += value;
            isFirst = true;
        } else if (!isNaN(value)) { // 숫자면 (공백도 숫자로 취급하므로 조심!)
            answer += value;
            isFirst = false;
        } else if (value === value.toUpperCase()) { // 대문자면
            answer += isFirst ? value : value.toLowerCase();
            isFirst = false;
        } else { // 소문자면
            answer += isFirst ? value.toUpperCase() : value;
            isFirst = false;
        }
    });
    
    return answer;
}